Classes/Repository/IssueSummaryType.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  IssueSummaryType.swift
+//  Freetime
+//
+//  Created by Sherlock, James on 29/07/2017.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+import IGListKit
+
+protocol IssueSummaryType {
+    
+    var id: String { get }
+    var attributedTitle: NSAttributedStringSizing { get }
+    var number: Int { get }
+    var createdAtDate: Date? { get }
+    var rawState: String { get }
+    var authorName: String? { get }
+    var isIssue: Bool { get }
+    
+}
+
+extension IssueSummaryType {
+    var stateIcon: UIImage? {
+        var name: String?
+        
+        switch rawState {
+        case IssueState.open.rawValue where isIssue: name = "issue-opened"
+        case IssueState.closed.rawValue where isIssue: name = "issue-closed"
+        case PullRequestState.open.rawValue where !isIssue: name = "git-pull-request"
+        case PullRequestState.closed.rawValue where !isIssue: name = "git-pull-request"
+        case PullRequestState.merged.rawValue where !isIssue: name = "git-merge"
+        default: return nil
+        }
+        
+        return UIImage(named: name ?? "")?.withRenderingMode(.alwaysTemplate)
+    }
+    
+    var stateColor: UIColor? {
+        var hex: String?
```

Comment:
Will the compiler complain when you make this a `let` and a `String` instead of a `String?` because of the `return nil`?